### ISSUE TITLE
Benchmark Fix for RandomForest

### ISSUE THREAD

**Issue by Wei-1** [sayakpaul/Adult-Dataset-Census-Dataset#1]
In the [benchmark file](Benchmarking_experiments_on_the_Adult_dataset_and_interpretability.ipynb)

We use the `LabelEncoder` on our training and testing data separately, which will cause index mismatch.

```python
from sklearn.preprocessing import LabelEncoder

train_preprocessed = train.apply(LabelEncoder().fit_transform)
test_preprocessed = test.apply(LabelEncoder().fit_transform)
```

We should change it with `OrdinalEncoder`

```python
from sklearn.preprocessing import OrdinalEncoder
oe = OrdinalEncoder(handle_unknown='use_encoded_value', unknown_value=-1)

train_preprocessed = oe.fit_transform(train.fillna("NaN"))
test_preprocessed = oe.transform(test.fillna("NaN"))
train_preprocessed = pd.DataFrame(train_preprocessed, columns=train.columns)
test_preprocessed = pd.DataFrame(test_preprocessed, columns=train.columns)
```

With this transformation, I am able to get: `0.8526` accuracy for Random Forest.
